### PR TITLE
fix(gui-v2): show form items in editable state and handle LTAR columns

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- Show form view inputs in an editable state
- Handle LTAR columns and show cards for has many and many to many.
- LTAR issues in the grid view
- Date/Year/Time/Datetime picker bugs

 re #3161 , #3164, #3144

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Screenshot 2022-08-16 at 11 31 29 AM
Screenshot 2022-08-16 at 11 31 39 AM

### CODE REVIEW

File: packages/nc-gui-v2/components/smartsheet/Form.vue [hunk 1]
```diff
@@ -22,9 +22,12 @@ import {
   useViewData,
   watch,
 } from '#imports'
+import { EditModeInj } from '~/context'
```
Comment (wingkwong) [inline packages/nc-gui-v2/components/smartsheet/Form.vue:25]: put it into `#imports`?

Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Form.vue:25]: -> `'#imports'`

File: packages/nc-gui-v2/components/smartsheet/Form.vue [hunk 2]
```diff
@@ -53,6 +56,13 @@ const { showAll, hideAll, saveOrUpdate } = useViewColumns(view, meta as any, fal
   setFormData()
 })
 
+
+const { syncLTARRefs } = useProvideSmartsheetRowStore(meta!, ref({
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Form.vue:60]: You could add the `!` marker at `inject(MetaInj)!`

File: packages/nc-gui-v2/components/smartsheet/Form.vue [hunk 3]
```diff
@@ -568,9 +578,11 @@ onMounted(async () => {
                     </a-form-item>
 
                     <div class="items-center flex">
-                      <span class="text-sm text-gray-500 mr-2 nc-form-input-required">{{ $t('general.required') }}</span>
+                      <span class="text-sm text-gray-500 mr-2 nc-form-input-required">{{ $t('general.required')
+                        }}</span>
 
-                      <a-switch v-model:checked="element.required" size="small" class="my-2" @change="updateColMeta(element)" />
+                      <a-switch v-model:checked="element.required" size="small" class="my-2"
+                                @change="updateColMeta(element)" />
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Form.vue:585]: formatting here seems slightly off

File: packages/nc-gui-v2/components/smartsheet/Form.vue [hunk 4]
```diff
@@ -653,7 +665,8 @@ onMounted(async () => {
               />
               <!-- Email me at <email> -->
               <span class="ml-4">
-                {{ $t('msg.info.emailForm') }} <span class="text-bold text-gray-600">{{ state.user.value?.email }}</span>
+                {{ $t('msg.info.emailForm') }} <span class="text-bold text-gray-600">{{ state.user.value?.email
+                }}</span>
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Form.vue:669]: Formatting

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by wingkwong — COMMENTED** [review]
1. Filter value should be cleared when the modal is closed
    - Steps to reproduce:
      - Click `Link to 'Actor'`
      - Search `camelot`
      - Click `CAMELOT VACATION`
      - Click `Link to 'Actor'`
      - Filtered result is shown while filter value is cleared in v1
2. Deleting a record causes a 404 error.
image

**Review by dstala — CHANGES_REQUESTED** [review]
- Reduce height of HM column in the form
- Cell that need picker are still not editable

**Comment by pranavxc** [comment]
> 1. Filter value should be cleared when the modal is closed
>    
>    * Steps to reproduce:
>      
>      * Click `Link to 'Actor'`
>      * Search `camelot`
>      * Click `CAMELOT VACATION`
>      * Click `Link to 'Actor'`
>      * Filtered result is shown while filter value is cleared in v1
> 2. Deleting a record causes a 404 error.
>    image

Fixed

**Comment by pranavxc** [comment]
> * Reduce height of HM column in the form
> * Cell that need picker are still not editable

Updated height and fixed cell related issues

**Review by bcakmakoglu — APPROVED** [review]

**Review by wingkwong — APPROVED** [review]

**Review by dstala — APPROVED** [review]

**Pull request merged**